sycl/source/detail/graph_impl.hpp

```diff
@@ -695,11 +695,17 @@ class graph_impl {
 
   // Returns the number of nodes in the Graph
   // @return Number of nodes in the Graph
-  size_t getNumberOfNodes() const {
+  size_t getNumberOfNodes() {
```

Comment:
I guess this was removed because we need to modify the visited flag in the nodes, but I think it would be fine to mark that flag as mutable and keep this const, since the flag doesn't meaningfully modify anything about the node.

Reply:
I think we can't make this function const even if we make MVisited mutable because this function calls searchDepthFirst which is a non-const function which therefore makes no promise not to modify the object.
And I don't think that we should make searchDepthFirst const since the caller-defined function called in searchDepthFirst could modify the nodes.

Reply:
OK that's fair!